hudi-common/src/main/java/org/apache/hudi/common/table/read/HoodiePositionBasedFileGroupRecordBuffer.java

```diff
@@ -199,7 +199,7 @@ public void processNextDeletedRecord(DeleteRecord deleteRecord, Serializable rec
     if (recordOpt.isPresent()) {
       String recordKey = recordOpt.get().getRecordKey();
       records.put(recordPosition, Pair.of(Option.empty(), readerContext.generateMetadataForRecord(
-          recordKey, recordOpt.get().getPartitionPath(), recordOpt.get().getOrderingValue(), orderingFieldType)));
+          recordKey, recordOpt.get().getPartitionPath(), recordOpt.get().getOrderingValue() == null ? orderingFieldDefault : recordOpt.get().getOrderingValue(), orderingFieldTypeOpt)));
```

Comment:
We should not set `orderingFieldDefault` which can cause confusion and written to the storage eventually.

Reply:
then what should we use as the ordering value when it is null?

Reply:
If it's null, we assume natural ordering.  Do we assume the ordering value to be always non-null in the record buffer?

Reply:
HUDI-8425 as a follow-up.